TeamCode/src/main/java/org/firstinspires/ftc/teamcode/trajectorysequence/TrajectorySequenceBuilder.java

```diff
@@ -549,46 +549,54 @@ private List<TrajectoryMarker> convertMarkersToGlobal(
     private List<SequenceSegment> projectGlobalMarkersToLocalSegments(List<TrajectoryMarker> markers, List<SequenceSegment> sequenceSegments) {
         if (sequenceSegments.isEmpty()) return Collections.emptyList();
 
+        markers.sort(Comparator.comparingDouble(TrajectoryMarker::getTime));
+
         double totalSequenceDuration = 0;
         for (SequenceSegment segment : sequenceSegments) {
             totalSequenceDuration += segment.getDuration();
         }
 
+        int segmentIndex = 0;
+        double currentTime = 0;
+
         for (TrajectoryMarker marker : markers) {
             SequenceSegment segment = null;
-            int segmentIndex = 0;
-            double segmentOffsetTime = 0;
 
-            double currentTime = 0;
-            for (int i = 0; i < sequenceSegments.size(); i++) {
-                SequenceSegment seg = sequenceSegments.get(i);
+            double markerTime = Math.min(marker.getTime(), totalSequenceDuration);
```

Comment:
Can we remove this now that we have a separate check? (And the code that computes `totalSequenceDuration`)

Reply:
Oh, yeah, sure!